gps/solution.go

```diff
@@ -50,16 +50,15 @@ type solution struct {
 
 const concurrentWriters = 16
 
-// WriteDepTree takes a basedir and a Lock, and exports all the projects
-// listed in the lock to the appropriate target location within the basedir.
+// WriteDepTree takes a basedir, a Lock and a RootPruneOptions and exports all
```

Comment:
Does this actually need a `RootPruneOptions`? shouldn't a `PruneOptions` be sufficient?

Reply:
Good catch. 😁

Reply:
hmm...ok, i think i see why this is a conundrum, and it connects with my above comment about perhaps rearranging and unexporting some of the types related to this.

could you maybe ping me tomorrow so we could voice about this a bit?